worker/sshserver/manifold.go

```diff
@@ -65,33 +70,22 @@ func (config ManifoldConfig) startWrapperWorker(context dependency.Context) (wor
 		return nil, errors.Trace(err)
 	}
 
-	var stTracker workerstate.StateTracker
-	if err := context.Get(config.StateName, &stTracker); err != nil {
-		return nil, errors.Trace(err)
-	}
-
-	statePool, err := stTracker.Use()
-	if err != nil {
-		_ = stTracker.Done()
-		return nil, errors.Trace(err)
-	}
-
-	sysState, err := statePool.SystemState()
+	controllerConfigWatcher, done, err := config.ControllerConfigWatcherGetter(context, config.StateName)
```

Comment:
If we ever want to use state again, and we follow this pattern, we'll need to return another done func per each usage I guess? Is there absolutely no way (other than what I did) to just get a state here and pass it through?

Reply:
If we need other state artefacts, we can tweak the func which uses state directly to return multiple results.